File: backend/internal/models/workspace.go
package models

import "time"

type Workspace struct {
	ID           string    `json:"id"`
	Name         string    `json:"name"`
	Path         string    `json:"path"`
	LastOpenedAt time.Time `json:"lastOpenedAt"`
	CreatedAt    time.Time `json:"createdAt"`
}

type WorkspaceCreate struct {
	Name              string `json:"name" binding:"required"`
	Path              string `json:"path" binding:"required"`
	OwnerDisplayName  string `json:"ownerDisplayName,omitempty"`
}

type WorkspaceUpdate struct {
	Name string `json:"name,omitempty"`
	Path string `json:"path,omitempty"`
}

File: frontend/src/shared/types/workspace.ts
export interface Workspace {
  id: string
  name: string
  path: string
  lastOpenedAt: string
  createdAt: string
}

export interface WorkspaceCreate {
  name: string
  path: string
  ownerDisplayName?: string
}

export interface WorkspaceUpdate {
  name?: string
  path?: string
}

export interface FileInfo {
  name: string
  path: string
  isDir: boolean
  size: number
  modTime: string
  mode: string
}

export interface BrowseResult {
  basePath: string
  home?: string
  files: FileInfo[]
}